src/debugConfigProvider.ts

```diff
@@ -145,11 +146,11 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
             delete dbgConfig.env;
             delete dbgConfig.args;
         }
-        else if (target.type === "Device")
-        {
-            let platformPath: string|void;
-            if (dbgConfig.iosRequest === "launch")
+        else if (target.type === "Device") 
             {
+            let platformPath: string|void;
+            if (dbgConfig.iosRequest === "launch") 
+                {
                 if (dbgConfig.iosInstallApp) {
                     platformPath = await targetCommand.deviceInstall({
```

Comment:
We'd also need to replace `deviceInstall`, right? Just noting, we can do this incrementally in a different PR as well.

Reply:
Installing the app still works with `ios-deploy`. Only getting the pid of a running process and spawning the debug server seems to require `DeveloperDiskImage.dmg` and hence is retired since iOS 17.
Of course long-term it would be sort of cleaner to use `xcrun devicectl` instead as well, but this will only work with XCode >= 15 installed, so at the time being `ios-deploy` actually provides the highest compatibility.